src/cobra/io/sbml.py

```diff
@@ -743,9 +742,6 @@ def process_association(ass):
                 if f_replace and F_GENE in f_replace:
                     gpr = " ".join(f_replace[F_GENE](t) for t in gpr.split(" "))
 
-        # remove outside parenthesis and format into standard form
-        gpr = ast2str(parse_gpr(gpr)[0])
-
         cobra_reaction.gene_reaction_rule = gpr
```

Comment:
Can't you just parse it into a GPR object directly and set `Reaction.gpr`? Should be much faster.

Reply:
I meant to not turn it to a string again and then assign to `cobra_reaction.gene_reaction_rule` because this just triggers another parsing. Rather do something like 

```python
gpr = parse_gpr(gpr)[0]
cobra_reaction.gpr = gpr
```

That only parses once.